src/lib/class/class.factory.ts

```diff
@@ -52,7 +55,14 @@ function transform(options: ClassOptions): ClassOptions {
 function generate(options: ClassOptions): Source {
   return (context: SchematicContext) =>
     apply(url(join('./files' as Path, options.language)), [
-      options.spec ? noop() : filter((path) => !path.endsWith('.spec.ts')),
+      options.spec
+        ? noop()
+        : filter((path) => {
+            const specNamingConvention = options.specFileSuffix || 'spec';
+            const languageExtension = options.language || 'ts';
```

Comment:
I didn't know if using a static file extension may result in a bug when using JS. Let me know if this is fine, and I'll revert this.